apps/pie-microsite/src/_11ty/shortcodes/tokensTable.js

```diff
@@ -130,6 +149,138 @@ const createTokensList = listElements => `<div class="c-tokensTable-row u-spacin
   ${listElements.join('')}
 </ul>`;
 
+/**
+ * Creates a tokens list for a given category
+ * @param {*} tokens 
+ * @param {string} path 
+ * @param {string} category 
+ * @param {string} tokenType 
+ * @returns 
+ */
+const buildTokensListForCategory = (tokens, path, category, tokenType) => {
+    const tokenTypeMetadata = getTokenTypeMetadata(path);
+    const tokensForCategory = getTokensForCategory(category, tokenTypeMetadata);
+
+    // create a list item for the current token
+    const tokenListElements = tokensForCategory.map(key => buildTokenListElements({
+        token: tokens[key],
+        tokenScssName: createScssTokenName(key, tokenType),
+        tokenDisplayName: createTokenDisplayName(key, tokenType),
+        tokenType,
+        tokenMetadata: tokenTypeMetadata[key]
+    }));
+
+    return buildTokensList(tokenListElements);
+};
+
+/**
+ * Builds a categorised list of tokens
+ * @returns {string}
+ */
+const buildCategorisedLists = ({
+    path, tokenType, tokens 
+}) => {
+    const categories = objectHelpers.getObjectPropertyByPath(pieTokenCategories, path);
+
+    // for each category, create an h2 and a list of token elements to render
+    const categoryTokenLists = Object.keys(categories).map(category => {
+        const heading = `<h2>${categories[category].displayName}</h2>`;
+        const tokensList = buildTokensListForCategory(tokens, path, category, tokenType);
+
+        // returns a 'chunk' of the tokens table page (a heading, the column headers, list of tokens and an option HR element)
+        return `${heading}${tokensList}`;
+    });
+
+    // all 'chunks' of the tokens table page HTML in a single string
+    return categoryTokenLists.join('<hr />');
+};
+
+/**
+ * Gets all the metadata associated with tokens of a given type such as colour.
+ * @param {string} path 
+ * @returns {object}
+ */
+const getTokenTypeMetadata = path => objectHelpers.getObjectPropertyByPath(pieTokensMetadata, path);
```

Comment:
As this file is getting pretty big, I would look to split it out into separate files where logical.

Looking over it, it seems like there are creation/build functions and functions that interact directly with the data/metadata, so I'd probably split it based on that as a start to separate out a bit.

Reply:
I've separated out the two types of functions. I think both of these will end up changing a fair amount as we add new token types so I haven't changed much at this point.